conbench/api/_google.py

```diff
@@ -219,6 +206,42 @@ def conclude_oidc_flow():
     return user_came_from_url, userinfo
 
 
+def parse_state_from_authorization_response(c: WebApplicationClient, url: str) -> str:
```

Comment:
Huh! :) On the one hand I can be happy to change this because I can totally choose to not care in this case.

But because it's a calm morning and I just meditated I feel like trying to put my opinion into words. Maybe for myself more than for others, it's an interesting exercise. Feel free to skip reading.

I think I do not generally agree with

> Single-character variables are a little hard to read

There certainly are cases when single-character variable names are dangerous and a bit difficult to deal with, especially when they are re-used and when their definition is rather distant from their usage. They can also be tiring, I know what you mean I think.

However, there are also other cases where they are even likable when used well, and can even make code more readable and look beautiful because they allow for focusing on the verbs more than on the subjects, grammatically speaking (if that makes sense). 

In many pieces of code there is a variable that takes front stage and represents a really important object within a rather local context. And this important entity is often very locally represented by this one character of text. Some examples that we often come across: `f` for file, `r` for reader, `w.` for writer, `s` for server, `c` for client, well :) Or `s` for stream, `r` for response, and so on. Oftentimes we use those in a Python context manager, but other times these one-character variables are also the arguments to _short_ functions.

Here, I think the reading distance between `c: WebApplicationClient` and `c.parse_request_uri_response(url)` is pretty short. That is, "what is `c`?" is quickly answered. More importantly, even `client` would not be able to convey the complexity of this kind of object anyway. If the argument was called `client` then the immediate question "what kind of `client` is this, exactly?" would come up, and the type annotation would again help a bit. That's also why -- mainly for readability -- I did use the type annotation here.

Maybe for completeness, two examples for (IMO) readable code:

```python
    # Create gipc pipe and expose the read end as `r` and the write end as `w`.
    with gipc.pipe() as (r, w):

        # Start child process for receiving messages. It inherits the standard
        # streams of this process and prints the received messages to stdout.
        p = gipc.start_process(target=child_process, args=(r, ))

        # Start greenlet (in the current, the parent, process). It periodically
        # sends a message to the child process through the pipe, via gipc's IPC.
        g = gevent.spawn(_writegreenlet, w)

        # Keep the pipe alive, and let the two entities communicate as long as
        # the shutdown procedure has not been invoked.
        while not shutdown:
            gevent.sleep(0.01)

        # Once we're here the shutdown procedure has been invoked. Terminate the
        # message sender greenlet. `kill()` always returns None; never raises an
        # exception.
        g.kill(block=True)

        print('Write greenlet terminated. Send SIGTERM to child process')
        p.terminate()

        # Wait for child process to terminate, reap it (read exit code).
        p.join()
        print('Child process terminated. Exit code: %s' % (p.exitcode, ))
```
(from [here](https://github.com/jgehrcke/gipc/blob/4f5195c1c830e4e8684c6b558fdad5dd89c7930f/examples/infinite_send_to_child.py#L77))


And then maybe this here from the Go stdlib docs:
```golang
s := &http.Server{
	Addr:           ":8080",
	Handler:        myHandler,
	ReadTimeout:    10 * time.Second,
	WriteTimeout:   10 * time.Second,
	MaxHeaderBytes: 1 << 20,
}
log.Fatal(s.ListenAndServe())
```

I should stop typing now. Hope this doesn't waste anyone's time. :)

Reply:
haha I really appreciate the thorough response! It's much more clear to me that this is a stylistic opinion that depends on diverse backgrounds and experiences reading text. Different people have different opinions of beauty.

My background: On previous teams I've been on, we tried to stay away from single-character arguments to functions, because we had lots of poor experience trying to understand unreadable code with non-intuitive or ill-defined variable names. So it's just something I've become accustomed to when reading code. For example, I still think those code blocks you attached could be improved with a few more characters. 😉  But I agree that I use them all the time for context managers. And my comment was just a `nit` so I love hearing your opinions :)

Reply:
<3